### PR TITLE
refactor(Table): Migrate to `styled-components`

### LINKED ISSUE #1903: Migrate `Table` to styled-components

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #1903

## Overview

Migrate `Table` to styled-components.

## Reason

Migrating all components.

### CODE REVIEW

**Comment by m7kvqbe1** [comment]
There were a few redundant styles I removed that appear to be left over for a historic refactor.

File: packages/react-component-library/src/components/Table/partials/StyledTable.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { selectors } from '@royalnavy/design-tokens'
+
+const { spacing, fontSize, color } = selectors
+
+export const StyledTable = styled.table`
+  table-layout: fixed;
+  width: 100%;
+  border-collapse: collapse;
+
+  thead {
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/Table/partials/StyledTable.tsx]: I wonder if we should have a `StyledTableHead`, `StyledTableRow` etc?

**Review by thyhjwb6 — COMMENTED** [review]
Looks good 👍 

There is a minor observation about whether to break up the styled table.

**Review by thyhjwb6 — APPROVED** [review]
👍

**Pull request merged**